client/dashboard/domains/dataviews/fields.tsx

```diff
@@ -118,57 +118,62 @@ export const useFields = ( {
 			},
 			{
 				id: 'expiry',
-				label: __( 'Expires/Renews on' ),
+				label: __( 'Paid until' ),
 				enableHiding: false,
 				enableSorting: true,
+				sort: ( a, b, direction ) => {
+					if ( a.expiry === null && b.expiry === null ) {
+						return 0;
+					}
+					if ( a.expiry === null ) {
+						return 1;
+					}
+
+					if ( b.expiry === null ) {
+						return -1;
+					}
+
+					const factor = direction === 'asc' ? 1 : -1;
+					return a.expiry.localeCompare( b.expiry ) * factor;
+				},
 				elements: [
-					{ value: 'next-7-days', label: __( 'Next 7 days' ) },
-					{ value: 'next-30-days', label: __( 'Next 30 days' ) },
-					{ value: 'next-90-days', label: __( 'Next 90 days' ) },
-					{ value: 'expired', label: __( 'Expired' ) },
+					{ value: '2-next-07-days', label: __( 'Next 7 days' ) },
+					{ value: '2-next-30-days', label: __( 'Next 30 days' ) },
+					{ value: '2-next-90-days', label: __( 'Next 90 days' ) },
+					{ value: '3-more-than-90-days', label: __( 'More than 90 days' ) },
+					{ value: '1-expired', label: __( 'Expired' ) },
+					{ value: '0-no-expiry', label: __( 'No expiry date' ) },
 				],
 				filterBy: {
 					operators: [ 'isAny' as Operator ],
 				},
 				getValue: ( { item }: { item: DomainSummary } ) => {
 					if ( ! item.expiry ) {
-						return '';
+						return '0-no-expiry';
 					}
 
 					const expiryDate = new Date( item.expiry );
 					const now = new Date();
-					const diffInDays = Math.ceil(
-						( expiryDate.getTime() - now.getTime() ) / ( 1000 * 60 * 60 * 24 )
-					);
+					const diffInMs = expiryDate.getTime() - now.getTime();
+					const diffInDays = Math.ceil( diffInMs / ( 1000 * 60 * 60 * 24 ) );
 
-					if ( diffInDays < 0 ) {
-						return 'expired';
-					} else if ( diffInDays <= 7 ) {
-						return 'next-7-days';
-					} else if ( diffInDays <= 30 ) {
-						return 'next-30-days';
+					if ( item.expired ) {
+						return '1-expired';
 					} else if ( diffInDays <= 90 ) {
-						return 'next-90-days';
+						return '2-next-90-days';
+					} else if ( diffInDays <= 30 ) {
+						return '2-next-30-days';
+					} else if ( diffInDays <= 7 ) {
+						return '2-next-07-days';
 					}
-
-					return 'later';
+					return '3-more-than-90-days';
```

Comment:
@hambai I think there is an issue with filters, but I'm not sure if it's fixable.
I have a domain paid until next 23 days. It doesn't show if a select "Next 30 days", but only if I select "Next 90 days". It works if I reverse the else checks (7, 30 and 90), but then it doesn't show anymore selecting "Next 90 days". I don't think the filter can work with overlapping values by design, but only with specific ranges (<7, >7 and <30, >30 and <90) which it's a bit weird.